smtp/message.js

```diff
@@ -309,34 +441,38 @@ class MessageStream extends Stream {
 			fs.open(attachment.path, 'r', opened);
 		};
 
+		/**
+		 * @param {MessageAttachment} attachment the metadata to use as headers
+		 * @param {function(): void} callback the function to call after output is finished
+		 * @returns {void}
+		 */
 		const output_stream = (attachment, callback) => {
 			if (attachment.stream.readable) {
-				let previous = null;
+				let previous = Buffer.alloc(0);
 
 				attachment.stream.resume();
 
 				attachment.stream.on('end', () => {
-					output_base64(
-						(previous || Buffer.from(0)).toString('base64'),
-						callback
-					);
+					output_base64(previous.toString('base64'), callback);
 					this.removeListener('pause', attachment.stream.pause);
 					this.removeListener('resume', attachment.stream.resume);
 					this.removeListener('error', attachment.stream.resume);
 				});
 
-				attachment.stream.on('data', buffer => {
+				attachment.stream.on('data', buff => {
 					// do we have bytes from a previous stream data event?
-					if (previous) {
+					let buffer = Buffer.isBuffer(buff) ? buff : Buffer.from(buff);
+
+					if (previous.byteLength > 0) {
 						const buffer2 = Buffer.concat([previous, buffer]);
-						previous = null; // free up the buffer
+						previous = Buffer.alloc(0); // free up the buffer
```

Comment:
would this need to be a `null` assignment to release the memory or is this safe?

Reply:
based off my reading of the current documentation, none of this is needed and the GC should figure it out as long as we don't have any pointers to the buffer hanging around.

Reply:
dropped in ff2ded902d2813bbfc6876103b977577e69f7976